### PR TITLE
Claim gateways on The Things Gateway Controller

### PR DESCRIPTION
**Pull request by johanstokking**
#### Summary

References https://github.com/TheThingsIndustries/lorawan-stack/issues/4256
Follow up on https://github.com/TheThingsNetwork/lorawan-stack/pull/7198 and https://github.com/TheThingsIndustries/lorawan-stack/issues/4301

#### Changes

This implements The Things Gateway Controller upstream for gateway claiming.

##### Steps

...

##### Regressions

...

#### Notes for Reviewers

This is draft until testing is done and testing steps are filled out. The code can already be reviewed.

The unit tests are quite sparse because it's a light API wrapper for The Things Gateway Controller. I don't think it makes a lot of sense to stub te server as there isn't a lot of logic that needs testing or protection against regressions. I'm also fine adding server stubs though.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [ ] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/deviceclaimingserver/gateways/gateways.go [hunk 1]
```diff
@@ -76,7 +87,7 @@ type Claimer interface {
 	// Claim claims a gateway.
 	Claim(ctx context.Context, eui types.EUI64, ownerToken string, clusterAddress string) error
 	// Unclaim unclaims a gateway.
-	Unclaim(context.Context, types.EUI64, string) error
+	Unclaim(ctx context.Context, eui types.EUI64) error
```
Comment (johanstokking) [inline pkg/deviceclaimingserver/gateways/gateways.go:90]: The third parameter was ambiguous: there was code that passed the cluster address and code that passed the owner token. I don't think we need either.

File: pkg/deviceclaimingserver/gateways/ttgc/ttgc.go [hunk 2]
```diff
@@ -16,35 +16,198 @@
...
+		return err
+	}
+	return nil
+}
+
+// IsManagedGateway implements gateways.GatewayClaimer.
+// This method always returns true.
+func (*Upstream) IsManagedGateway(context.Context, types.EUI64) (bool, error) {
+	return true, nil
 }
```
Comment (KrishnaIyer) [inline pkg/deviceclaimingserver/gateways/ttgc/ttgc.go:213]: This is fine for now but when (if) we add CUPS support to the GC, we need to make this dependent on the EUI range.

**Review by KrishnaIyer — APPROVED** [review]
LGTM

File: pkg/deviceclaimingserver/gateways/ttgc/ttgc.go (hunk 2 above)
Comment (johanstokking) [reply pkg/deviceclaimingserver/gateways/ttgc/ttgc.go:213]: That's why there's a `types.EUI64` argument and this method returns an error: we can differentiate EUI ranges (based on config) and even make an rpc call to figure this out.

**Pull request merged**